### DISCUSSION TITLE
Porting from go-libp2p to rust-libp2p, I need help with the gossipsub implementation

### DISCUSSION THREAD

**Discussion by harsh4786** [libp2p/rust-libp2p#4417] [Q&A]
It's me again, im trying to use rust libp2p gossipsub, is there a way to do `envelope, err := sub.Next(ctx)` where `sub = th.Subscribe(pubsub.WithBufferSize(P2P_SUBSCRIPTION_BUFFER_SIZE))`, as you can see this is in Go, I want to see if i can iterate over gossipsub subscription stream in rust.

**Comment by thomaseizinger** [comment]
The API of gossipsub is documented here: https://docs.rs/libp2p-gossipsub/0.45.1/libp2p_gossipsub/struct.Behaviour.html#implementations

Subscribing returns you a bool whether or not you've been subscribed: https://docs.rs/libp2p-gossipsub/0.45.1/libp2p_gossipsub/struct.Behaviour.html#method.subscribe

Once you are subscribed, you will receive events: https://docs.rs/libp2p-gossipsub/0.45.1/libp2p_gossipsub/enum.Event.html

`rust-libp2p` works a bit differently than `go-libp2p`. Checkout the chat example: https://github.com/libp2p/rust-libp2p/tree/master/examples/chat-example
